packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts

```diff
@@ -178,6 +193,51 @@ export class FunctionTransformer extends TransformerPluginBase {
   };
 }
 
+/**
+ * The response mapping template for 'RequestResponse' (default) and 'Event' invocation types differ.
+ * Use this to generate the appropriate response mapping template.
+ * @param config the {@link FunctionDirectiveConfiguration} for `@function` definition.
+ * @returns the response mapping template used by AppSync to handle responses from the Lambda function invocation.
+ */
+const responseMappingTemplate = (config: FunctionDirectiveConfiguration): string => {
+  if (config.invocationType === 'Event') {
+    /*
+      #set( $success = true )
+      #if( $ctx.error )
+        $util.appendError($ctx.error.message, $ctx.error.type)
+        #set( $success = false )
+      #end
+      #set( $response = {
+        "success": $success
+      } )
+      $util.toJson($response)
+    */
+    return printBlock('Handle error or return result')(
+      compoundExpression([
+        set(ref('success'), bool(true)),
+        iff(
+          ref('ctx.error'),
+          compoundExpression([raw('$util.appendError($ctx.error.message, $ctx.error.type)'), set(ref('success'), bool(false))]),
+        ),
+        compoundExpression([set(ref('response'), obj({ success: ref('success') })), raw('$util.toJson($response)')]),
+      ]),
+    );
+  }
+
+  /*
+    #if( $ctx.error )
+      $util.error($ctx.error.message, $ctx.error.type)
+    #end
+    $util.toJson($ctx.result)
+  */
```

Comment:
nit for future PR: remove or explain

Reply:
I find these comment blocks helpful to understand the actual request/response mapping templates we're generating (much easier to read than the VTL helpers) within the context of the transformer logic, i.e. without having to peruse test cases to find applicable snapshots. But I'm fine removing them if you feel they just add noise.